integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/listeners/PreAggregateRenameTablePostListener.scala

```diff
@@ -0,0 +1,133 @@
+    implicit val sparkSession: SparkSession = renameTablePostListener.sparkSession
+    val renameTableModel = renameTablePostListener.alterTableRenameModel
+    val oldParentTableName = renameTableModel.oldTableIdentifier.table
+    val oldParentDatabaseName = renameTableModel.oldTableIdentifier.database.getOrElse("default")
+    val newParentTableName = renameTableModel.newTableIdentifier.table
+    if (carbonTable.hasPreAggregateTables) {
+      val dataMapSchemas = carbonTable.getTableInfo.getDataMapSchemaList.asScala
+      dataMapSchemas.foreach {
+        dataMapSchema =>
+          val childTableIdentifier = dataMapSchema.getRelationIdentifier
+          val childCarbonTable = PreAggregateUtil
+            .getChildCarbonTable(childTableIdentifier.getDatabaseName,
+              childTableIdentifier.getTableName)(sparkSession)
+          updateChildTableWithNewParent(renameTableModel, childCarbonTable)(sparkSession)
+      }
+    }
+    val schemaEvolutionEntry = new SchemaEvolutionEntry(System.currentTimeMillis)
+    schemaEvolutionEntry.setTime_stamp(System.currentTimeMillis())
+    updateParentRelationIdentifierForColumns(carbonTable.getTableInfo,
+      oldParentDatabaseName,
+      oldParentTableName,
+      newParentTableName)
+    updateTableSchema(carbonTable,
+      schemaEvolutionEntry,
+      oldParentDatabaseName,
+      oldParentTableName)
+  }
+
+  private def updateChildTableWithNewParent(renameTableModel: AlterTableRenameModel,
+      childCarbonTable: Option[CarbonTable])(implicit sparkSession: SparkSession): Unit = {
+    val oldParentTableName = renameTableModel.oldTableIdentifier.table
+    val oldParentDatabaseName = renameTableModel.oldTableIdentifier.database.getOrElse("default")
+    val newParentTableName = renameTableModel.newTableIdentifier.table
+    if (childCarbonTable.isDefined) {
+      val schemaEvolutionEntry = new SchemaEvolutionEntry(System.currentTimeMillis)
+      schemaEvolutionEntry.setTime_stamp(System.currentTimeMillis())
+      val childTableInfo = childCarbonTable.get.getTableInfo
+      childTableInfo.getParentRelationIdentifiers.asScala.foreach {
+        parentRelationIdentifier =>
+          if (parentRelationIdentifier.getDatabaseName
+                .equalsIgnoreCase(oldParentDatabaseName) &&
+              parentRelationIdentifier.getTableName
+                .equalsIgnoreCase(oldParentTableName)) {
+            parentRelationIdentifier.setTableName(newParentTableName)
+          }
+      }
+      updateTableSchema(childCarbonTable.get,
+        schemaEvolutionEntry,
+        childTableInfo.getDatabaseName,
+        childTableInfo.getFactTable.getTableName)
+    }
+  }
+
+  private def updateParentRelationIdentifierForColumns(parentTableInfo: TableInfo,
+      oldParentDatabaseName: String,
+      oldParentTableName: String,
+      newParentTableName: String): Unit = {
+    parentTableInfo.getFactTable.getListOfColumns.asScala.foreach {
+      columnSchema =>
```

Comment:
move to above line

Reply:
@ravipesala As discussed with vishal, rename table has a lot ol places that need to be updated with new parent name.

1. All the child tables have to be updated.
2. All parentColumnIdentifiers have to be updated with the new table
3. table properties map has to be updated for all childs.
4. the select query has to be updated with new parent for all childs.

Is it possible to block rename too?

Reply:
Ok, please block it for now.